### ISSUE TITLE
Replace dependency on dartsass-ruby with sass-embedded

### ISSUE THREAD

**Issue by johnnyshields** [tablecheck/dartsass-sprockets#2]
As per [this thread](https://github.com/sass/sassc-ruby/issues/220#issuecomment-1416158475), the [dartsass-ruby](tablecheck/dartsass-ruby) gem is a compatibility shim which provides legacy SassC features on Dart Sass.

When Dart Sass 2.0 is released, many of these features will become impossible to support.

Therefore, we should get rid of the dartsass-ruby dependency and replace it with a direct integration to [sass-embedded](https://github.com/ntkme/sass-embedded-host-ruby/issues). This involves generally replacing any references to the `::SassC` namespace. The sass-embedded documentation is [here](https://rubydoc.info/gems/sass-embedded/Sass).

To recap:

* Current dependency pattern: [dartsass-sprockets](https://github.com/tablecheck/dartsass-sprockets) --> [dartsass-ruby](tablecheck/dartsass-ruby) --> [sass-embedded](https://github.com/ntkme/sass-embedded-host-ruby/issues)

* Future dependency pattern: [dartsass-sprockets](https://github.com/tablecheck/dartsass-sprockets) --> [sass-embedded](https://github.com/ntkme/sass-embedded-host-ruby/issues)

A PR would be welcome for this.

**Comment by johnnyshields** [comment]
This is done
